### PR TITLE
[WIP] Adapt serverless docs for the new KafkaChannel implementation

### PR DESCRIPTION
**Pull request by aliok**
Version(s):

Issue:

Link to docs preview:

Additional information:

- Change serverless docs for the new KafkaChannel implementation

- Basically we have these differences:
   - Dataplane looks different (docs updated in this PR)
   - Secrets need to be set differently (docs updated in this PR)

What's not mentioned in this PR and we need to mention:
- Why we have a new KafkaChannel implementation (optional content)
- Serverless operator is providing an automated migration from the old configmap format to new configmap format.
- For the required new keys in the secret (`protocol` and `sasl.mechanism`), we are falling back to the old configmap and secret format and infer the values for these keys. However, this fallback code and inferring will be removed in the future. Thus, after the upgrade, our users should update their secrets to have these 2 keys.

Related upstream docs: https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/docs/channel/README.md

Upstream docs talk a lot about installation and migration. That's not relevant to downstream as the serverless operator takes care of those.

What could be relevant though:
- [Dataplane configuration](https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/docs/channel/README.md#configuring-dataplane): we currently don't have any docs around configuring the dataplane for the old/existing KafkaChannel
- [Discontinued features](https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/docs/channel/README.md#discontinued-features): Was never documented downstream anyway

### CODE REVIEW

File: modules/serverless-kafka-sasl-public-certs.adoc [hunk 1]
```diff
@@ -11,8 +11,9 @@ If you want to use SASL with public CA certificates, you must use the `tls.enabl
 [source,terminal]
 ----
 $ oc create secret --namespace <namespace> generic <kafka_auth_secret> \
+  --from-literal=protocol="SASL_PLAINTEXT" \
   --from-literal=tls.enabled=true \
   --from-literal=password="SecretPassword" \
-  --from-literal=saslType="SCRAM-SHA-512" \
+  --from-literal=sasl.mechanism="SCRAM-SHA-512" \
   --from-literal=user="my-sasl-user"
 ----
```
Comment (aliok) [inline modules/serverless-kafka-sasl-public-certs.adoc:19]: We now ask for a new key called `protocol` in the secret.
Also, key `saslType` is renamed to `sasl.mechanism`.

File: modules/serverless-kafka-sasl.adoc [hunk 2]
```diff
@@ -27,15 +27,21 @@ Red Hat recommends to enable TLS in addition to SASL.
...
 +
 [IMPORTANT]
 ====
-Use the key names `ca.crt`, `password`, and `saslType`. Do not change them.
+Use the key names `protocol`, `ca.crt`, `password`, and `sasl.mechanism`. Do not change them.
+====
+
+[NOTE]
+====
+If TLS is not required, do not specify the `ca.crt` key and change the `protocol` to `SASL_PLAINTEXT`.
```
Comment (aliok) [inline modules/serverless-kafka-sasl.adoc:44]: This explains the sentence here :https://github.com/openshift/openshift-docs/pull/46416/files#diff-2d568646d07f4762ad836f963b26119cfb38d2d4220bd42f75d6e457118dc18dR15

File: serverless/admin_guide/serverless-kafka-admin.adoc [hunk 3]
```diff
@@ -26,7 +26,7 @@ endif::[]
 The `KnativeKafka` CR provides users with additional options, such as:
 
 * Kafka source
-* Kafka channel
+* Kafka channel (Technology Preview)
```
Comment (aliok) [inline serverless/admin_guide/serverless-kafka-admin.adoc:29]: Are we back in TP with the KafkaChannel now?

@matzew @nainaz

**Comment by aliok** [comment]
cc @abrennan89 @Preeticp

File: modules/serverless-kafka-sasl.adoc (hunk 2 above)
Comment (abrennan89) [reply modules/serverless-kafka-sasl.adoc:44]: There shouldn't be two notes next to each other like this, it should probably be added as a callout if it's explaining something in the code block above?

File: modules/serverless-kafka-sasl-public-certs.adoc (hunk 1 above)
Comment (abrennan89) [reply modules/serverless-kafka-sasl-public-certs.adoc:19]: re the comment in the description:

> For the required new keys in the secret (protocol and sasl.mechanism), we are falling back to the old configmap and secret format and infer the values for these keys. However, this fallback code and inferring will be removed in the future. Thus, after the upgrade, our users should update their secrets to have these 2 keys.

Does this mean this needs to change again from the change shown here once a user upgrades to the new implementation?
In this case, I think we should just document this as a step as part of a document / release note about upgrading to the new version and publish that at the same time as any final changes here to the docs.

File: modules/serverless-kafka-sasl.adoc [hunk 4]
```diff
@@ -27,15 +27,21 @@ Red Hat recommends to enable TLS in addition to SASL.
 [source,terminal]
 ----
 $ oc create secret --namespace <namespace> generic <kafka_auth_secret> \
+  --from-literal=protocol="SASL_SASL" \
   --from-file=ca.crt=caroot.pem \
   --from-literal=password="SecretPassword" \
-  --from-literal=saslType="SCRAM-SHA-512" \
+  --from-literal=sasl.mechanism="SCRAM-SHA-512" \
   --from-literal=user="my-sasl-user"
 ----
```
Comment (abrennan89) [inline modules/serverless-kafka-sasl.adoc:35]: same question as above

File: modules/serverless-kafka-sasl.adoc [hunk 5]
```diff
@@ -27,15 +27,21 @@ Red Hat recommends to enable TLS in addition to SASL.
...
 +
 [IMPORTANT]
 ====
-Use the key names `ca.crt`, `password`, and `saslType`. Do not change them.
+Use the key names `protocol`, `ca.crt`, `password`, and `sasl.mechanism`. Do not change them.
```
Comment (abrennan89) [inline modules/serverless-kafka-sasl.adoc:39]: Will this be a breaking change, i.e. if a user has already completed this procedure with the old key names, do they need to rename these as part of the upgrade?

**Review by abrennan89 — CHANGES_REQUESTED** [review]

**Comment by abrennan89** [comment]
@aliok if the dataplane docs need to be added, can you please provide a draft procedure for what a user needs to do here and _why_. There is a full example of the configmap upstream but I doubt that _all_ of this is relevant to the docs here.

> Why we have a new KafkaChannel implementation (optional content)

This should not be _optional_ content if it's a big, user-facing change. I can see some points about why in the upstream docs, but it's not clear to me from an end user perspective why. Maybe this doesn't need to be included if the _whys_ are only relevant from an implementation detail / eng perspective?

**Comment by abrennan89** [comment]
@aliok once the questions are answered, let's please close this PR and use https://github.com/openshift/openshift-docs/pull/46632 to make all the changes at once.

**Comment by aliok** [comment]
> @aliok if the dataplane docs need to be added, can you please provide a draft procedure for what a user needs to do here and why. There is a full example of the configmap upstream but I doubt that all of this is relevant to the docs here.

We didn't document configuring the dataplane for the previous channel implementation downstream. That's why I didn't write anything about it.

> This should not be optional content if it's a big, user-facing change. I can see some points about why in the upstream docs, but it's not clear to me from an end user perspective why. Maybe this doesn't need to be included if the whys are only relevant from an implementation detail / eng perspective?

It is an implementation detail. API is the same. However, certain configuration options are being changed. Still, I wouldn't call this a user-facing change because users will not even notice anything thanks to Serverless Operator and the migration code we have.

@abrennan89

**Comment by abrennan89** [comment]
Closing; these details are now implemented in https://github.com/openshift/openshift-docs/pull/46632

**Pull request closed without merging**